### PR TITLE
Redirect blue site user to red site's login page

### PR DESCRIPTION
**Pull request by asadmanzoor93**
**Description:**  This PR adds the implementation to redirect one site user to other site’s logout page if one site user tries to access any other site’s LMS/Studio URL’s

**JIRA:** 
https://edlyio.atlassian.net/browse/EDLY-1380

**Merge checklist:**

- [ ] All reviewers approved
- [ ] Commits are (reasonably) squashed

**Post merge:**
- [ ] Delete working branch (if not needed anymore)

**Note:** Please add screenshots for any viewable change.
Screenshot 2020-06-01 at 5 11 31 PM

### CODE REVIEW

**Comment by asadmanzoor93** [comment]
> Kindly update the Title. It seems like it only takes a user from blue site to red. Change it to something like Redirect logged in user from another site to current site's login page
It was same in ticket title, however, I will update the description to remove any misunderstanding.

File: openedx/features/edly/middleware.py [hunk 1]
```diff
@@ -2,10 +2,12 @@
...
 """
 from logging import getLogger
+
 from django.conf import settings
+from django.contrib.auth import logout
 from django.contrib.sites.shortcuts import get_current_site
-from django.http import Http404
-
+from django.http import HttpResponseRedirect
+from django.urls import reverse
```
Comment (zubair-arbi) [inline openedx/features/edly/middleware.py:10]: nit: add a new line before local imports

File: openedx/features/edly/tests/test_middleware.py [hunk 2]
```diff
@@ -61,10 +62,11 @@ def test_user_without_edly_organization_access(self):
...
 
         with LogCapture(LOGGER_NAME) as logger:
-            response = self.client.get('/', follow=True)
-            assert response.status_code == 404
+            self.client.get('/', follow=True)
+            user = auth.get_user(self.client)
+            assert not user.is_authenticated()
 
-            logger.check(
+            logger.check_present(
```
Comment (zubair-arbi) [inline openedx/features/edly/tests/test_middleware.py:69]: we also need to check if user is correctly redirected after logout.
We can use `redirect_chain`

https://docs.djangoproject.com/en/3.0/topics/testing/tools/
https://stackoverflow.com/questions/14951356/django-testing-if-the-page-has-redirected-to-the-desired-url

Comment (zubair-arbi) [reply openedx/features/edly/tests/test_middleware.py:76]: Also update the docstring of this test case accordingly.

Comment (asadmanzoor93) [reply openedx/features/edly/tests/test_middleware.py:76]: Done

File: openedx/features/edly/tests/test_middleware.py [hunk 3]
```diff
@@ -54,17 +55,25 @@ def test_user_with_edly_organization_access(self):
...
-        Test that logged in user gets 404 and valid log message if user has no access for
+        Test that logged in user gets redirected to logout page and valid log message if user has no access for
         request site's edly sub organization.
         """
 
         with LogCapture(LOGGER_NAME) as logger:
             response = self.client.get('/', follow=True)
-            assert response.status_code == 404
+            self.assertRedirects(
+                response, '/logout',
```
Comment (zubair-arbi) [inline openedx/features/edly/tests/test_middleware.py:67]: nit: either keep all argument in one line or only one argument in one line.  https://docs.djangoproject.com/en/3.0/topics/testing/tools/#django.test.SimpleTestCase.assertRedirects
```python
response, 
'/logout',
```

Comment (asadmanzoor93) [reply openedx/features/edly/tests/test_middleware.py]: Done

**Pull request merged**